AccessibilityInsightsForAndroidService/app/src/main/java/com/microsoft/accessibilityinsightsforandroidservice/AccessibilityEventDispatcher.java

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package com.microsoft.accessibilityinsightsforandroidservice;
+
+import android.view.accessibility.AccessibilityEvent;
+import android.view.accessibility.AccessibilityNodeInfo;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class AccessibilityEventDispatcher {
+  private static final String TAG = "AccessibilityEventDispa";
+  private CharSequence previousPackageName;
+
+  private ArrayList<Consumer<AccessibilityNodeInfo>> onAppChangedListeners;
+  private ArrayList<Consumer<AccessibilityEvent>> onFocusEventListeners;
+  private ArrayList<Consumer<AccessibilityEvent>> onRedrawEventListeners;
+
+  public AccessibilityEventDispatcher() {
+    onAppChangedListeners = new ArrayList<Consumer<AccessibilityNodeInfo>>();
+    onFocusEventListeners = new ArrayList<Consumer<AccessibilityEvent>>();
+    onRedrawEventListeners = new ArrayList<Consumer<AccessibilityEvent>>();
+  }
+
+  public static List<Integer> redrawEventTypes =
+      Arrays.asList(
+          AccessibilityEvent.TYPE_WINDOW_STATE_CHANGED,
+          AccessibilityEvent.TYPE_VIEW_SCROLLED,
+          AccessibilityEvent.TYPE_WINDOW_CONTENT_CHANGED,
+          AccessibilityEvent.TYPE_WINDOWS_CHANGED);
+
+  public void onAccessibilityEvent(AccessibilityEvent event, AccessibilityNodeInfo rootNode) {
+    int eventType = event.getEventType();
```

Comment:
I assume this onAccessibilityEvent method is what will be called from the AccessibilityInsightsForAndroidService.onAccessibility event, right?  With that assumption in mind, 2 questions:

- We're currently calling event.getEventType() in AccessibilityInsightsForAndroidService.onAccessibilityEvent.  Will this just move that call into this class, or will we be calling it twice?  

- It's my understanding that to get the AccessibilityNodeInfo you have to call event.getEventSource().  Will we be calling that on the the main service class to pass it in? Is there a reason these 2 things are happening in different places?

Reply:
You're correct on both assumptions. The reason I did not include those changes in the PR was because that is likely going to be a larger refactor. This way, as a part of this feature, we only add the onAccessibilityEvent method call to the service and move on with what we need to do.

Then we circle back to refactor that into this later.